src/main/java/com/khalsa_ji/ems/rest/Get.java

```diff
@@ -0,0 +1,231 @@
+                    .setReportingTo(EmployeeBrief.convertFrom(reportingTo))
+                    .build();
+
+            list.add(chart);
+        }
+        return ResponseEntity.ok(list);
+    }
+
+    /**
+     * Method to fetch an instance of the {@code EmployeeChart} class derived from
+     * an instance of the {@code Employee} class specified by the given employee id
+     *
+     * @param ID Employee's id
+     * @return An instance of the {@code EmployeeChart} class
+     * @see Employee
+     * @see EmployeeChart
+     */
+
+    @GetMapping(value = "/employees/{ID}", produces = "application/json")
+    @ApiOperation(value = "Fetch detailed employee chart for the given Employee ID")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully fetched the required detailed employee chart"),
+            @ApiResponse(code = 400, message = "Employee ID could not be negative."),
+            @ApiResponse(code = 404, message = "Employee with the given Employee ID not found!")
+    })
+    public ResponseEntity<Object> getEmployee(@PathVariable(value = "ID") Long ID) {
+        //  Validating ID
+        switch(Validator.validateID(ID)) {
+            case nullNumber:        return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.nullNumber", EMPLOYEE, Locale.getDefault()));
+            case zero:              return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.zero", EMPLOYEE, Locale.getDefault()));
+            case negativeNumber:    return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.negativeNumber", EMPLOYEE, Locale.getDefault()));
+        }
+
+        Employee employee = service.getEmployeeByID(ID);
+
+        //  If employee with specified id does not exists.
+        if(employee.getEmployeeID() == 0)   return ResponseEntity.notFound().build();
+
+        Employee manager = service.getEmployeeByID(employee.getManagerID());
+        List<Employee> colleagues = service.getEmployeesByManagerID(employee.getManagerID());
+        List<Employee> reportingTo = service.getEmployeesByManagerID(employee.getEmployeeID());
+
+        //  Removing the current employee from the colleagues list.
+        colleagues.removeIf(colleague -> colleague.getEmployeeID() == employee.getEmployeeID());
+
+        //  Ordering the list by designation's level id and employee's name
+        colleagues.sort(ComparatorClass.customComparator);
+        reportingTo.sort(ComparatorClass.customComparator);
+
+        EmployeeChart chart = new EmployeeChartBuilder()
+                .setEmployee(EmployeeBrief.convertFrom(employee))
+                .setManager(EmployeeBrief.convertFrom(manager))
+                .setColleagues(EmployeeBrief.convertFrom(colleagues))
+                .setReportingTo(EmployeeBrief.convertFrom(reportingTo))
+                .build();
```

Comment:
You can  create method from this class, something like generateEmployeeChart(Employee employee), as this same lines of code has been used by both getEmployee() and getEmployeeDetails().
So  we can have better code usability.

Reply:
Sure, we could have a static method in the class EmployeeChart, which would be responsible of building the required object. And this method would take only a single parameter as you said, Employee class object and remaining fields such as manager, colleagues, subordinates could be derived from it.
Really a better and cleaner way to do things. It could also help in updating the same piece of code in other places such as in Put.java, etc.